internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1365 @@
+	VlanId             string `tfschema:"vlan_id"`
+}
+
+type InfrastructureNetworkModel struct {
+	DhcpEnabled bool          `tfschema:"dhcp_enabled"`
+	SubnetMask  string        `tfschema:"subnet_mask"`
+	Gateway     string        `tfschema:"gateway"`
+	IpPool      []IpPoolModel `tfschema:"ip_pool"`
+	DnsServer   []string      `tfschema:"dns_server"`
+}
+
+type IpPoolModel struct {
+	StartingAddress string `tfschema:"starting_address"`
+	EndingAddress   string `tfschema:"ending_address"`
+}
+
+type OptionalServiceModel struct {
+	CustomLocation string `tfschema:"custom_location"`
+}
+
+type PhysicalNodeModel struct {
+	Name        string `tfschema:"name"`
+	Ipv4Address string `tfschema:"ipv4_address"`
+}
+
+type StorageModel struct {
+	ConfigurationMode string `tfschema:"configuration_mode"`
+}
+
+func (StackHCIDeploymentSettingResource) ModelObject() interface{} {
+	return &StackHCIDeploymentSettingModel{}
+}
+
+func (StackHCIDeploymentSettingResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"stack_hci_cluster_id": commonschema.ResourceIDReferenceRequiredForceNew(&deploymentsettings.ClusterId{}),
+
+		"arc_resource_ids": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: machines.ValidateMachineID,
+			},
+		},
+
+		"version": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$`),
+				"the version must be a set of numbers separated by dots: `10.0.0.1`",
+			),
+		},
+
+		"scale_unit": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"adou_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"cluster": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"name": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringMatch(
+										regexp.MustCompile("^[a-zA-Z0-9-]{3,15}$"),
+										"the cluster name must be 3-15 characters long and contain only letters, numbers and hyphens",
+									),
+								},
+
+								"azure_service_endpoint": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+
+								"cloud_account_name": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
and here

Reply:
This should be storage account name, validation added.